controllers/articleController.js

```diff
@@ -165,11 +163,45 @@ const reactToArticle = async (req, res) => {
   }
 };
 
+// method for averaging all ratings of an article and
+// adding to the article object
+const addRatingsToArticle = (res, req, rated) => {
+  if (rated) {
+    return Rating.findAll({ where: { articleId: req.article.id } })
+      .then((ratings) => {
+        let ratingSum = 0;
+        ratings.forEach((rate) => { ratingSum += rate.starRating; });
+        const averageRating = ratingSum / ratings.length;
+
+        return req.article.update({ averageRating })
+          .then(ratedArticle => res.status(201).send({
+            message: `You've rated this article ${req.userRating} star`,
+            ratedArticle,
+            ratings
+          }));
+      });
+  }
+  return res.status(500).send({
+    error: 'Internal server error'
+  });
+};
+
```

Comment:
Hi Victor, this looks great. However, I think we should use the 'async await' pattern whenever possible. Also, I would suggest using the 'try and catch' pattern as well.